packages/react-router/docs/api/Route.md

```diff
@@ -69,7 +69,17 @@ const User = ({ match }) => {
 }
 ```
 
-When you use `component` (instead of `render` or `children`, below) the router uses [`React.createElement`](https://facebook.github.io/react/docs/react-api.html#createelement) to create a new [React element](https://facebook.github.io/react/docs/rendering-elements.html) from the given component. That means if you provide an inline function to the `component` prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component. When using an inline function for inline rendering, use the `render` or the `children` prop (below).
+When you use `component` (instead of `render` or `children`, below) the router uses [`React.createElement`](https://facebook.github.io/react/docs/react-api.html#createelement) to create a new [React element](https://facebook.github.io/react/docs/rendering-elements.html) from the given component.
+
+That means if you provide an inline function to the `component` prop, you would create a new component every render. This results in the existing component unmounting and the new component mounting instead of just updating the existing component. When using an inline function for inline rendering, use the `render` or the `children` prop (below).
+
+```js
+<Route path="/user/:username" component={(props) => <User {...props} />}/>
```

Comment:
I don't think that the documentation should show `<Route component={() => {...}} />` anywhere. If you need to force React to create a new component, adding a `key` prop is a much better approach.